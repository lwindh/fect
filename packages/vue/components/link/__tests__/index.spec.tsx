import Link from '../index'
import { mount } from '@vue/test-utils'

describe('Link', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Link)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should be render correctly', async () => {
    const wrapper = mount({
      render() {
        return (
          <>
            <Link href="https://www.test.com">Link</Link>
            <Link href="https://www.test.com" color>
              Link
            </Link>
            <Link href="https://www.test.com" underline>
              Link
            </Link>
            <Link href="https://www.test.com" color underline>
              Link
            </Link>
            <Link href="https://www.test.com" block>
              Link
            </Link>
          </>
        )
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('props to should higher than props href', async () => {
    const wrapper = mount(Link, {
      props: {
        to: '#',
      },
    })
    const el = wrapper.find('.fect-link')
    await el.trigger('click')
    expect(() => wrapper.unmount()).not.toThrow()
  })
})
