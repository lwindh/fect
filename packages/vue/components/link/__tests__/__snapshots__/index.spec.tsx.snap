// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should be render as a element 1`] = `
<a class="fect-link " href="">
  <!---->
</a>
`;

exports[`Link should be render correctly 1`] = `<a class="fect-link " href="https://www.test.com">Link</a><a class="fect-link color" href="https://www.test.com">Link</a><a class="fect-link underline" href="https://www.test.com">Link</a><a class="fect-link color underline" href="https://www.test.com">Link</a><a class="fect-link block" href="https://www.test.com">Link</a>`;
